pkg/exporter/convert_flp_test.go

```diff
@@ -0,0 +1,376 @@
+			flow: &flow.Record{
+				RawRecord: flow.RawRecord{
+					Id: ebpf.BpfFlowId{
+						EthProtocol:       2048,
+						Direction:         flow.DirectionEgress,
+						SrcMac:            flow.MacAddr{0x04, 0x05, 0x06, 0x07, 0x08, 0x09},
+						DstMac:            flow.MacAddr{0x0a, 0x0b, 0x0c, 0x0d, 0x0e, 0x0f},
+						SrcIp:             flow.IPAddr{0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0xff, 0xff, 0x06, 0x07, 0x08, 0x09},
+						DstIp:             flow.IPAddr{0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0xff, 0xff, 0x0a, 0x0b, 0x0c, 0x0d},
+						SrcPort:           23000,
+						DstPort:           443,
+						TransportProtocol: 6,
+					},
+					Metrics: ebpf.BpfFlowMetrics{
+						Bytes:   456,
+						Packets: 123,
+						Flags:   0x100,
+						Dscp:    64,
+						PktDrops: ebpf.BpfPktDropsT{
+							Packets:         10,
+							Bytes:           100,
+							LatestFlags:     0x200,
+							LatestState:     6,
+							LatestDropCause: 5,
+						},
+						DnsRecord: ebpf.BpfDnsRecordT{
+							Latency: uint64(someDuration),
+							Id:      1,
+							Flags:   0x8001,
+							Errno:   0,
+						},
+					},
+				},
+				Interface:     "eth0",
+				TimeFlowStart: someTime,
+				TimeFlowEnd:   someTime,
+				AgentIP:       net.IPv4(0x0a, 0x0b, 0x0c, 0x0d),
+				DNSLatency:    someDuration,
+				TimeFlowRtt:   someDuration,
+			},
+			expected: &config.GenericMap{
+				"FlowDirection":          1,
+				"Bytes":                  456,
+				"SrcAddr":                "6.7.8.9",
+				"DstAddr":                "10.11.12.13",
+				"Dscp":                   64,
+				"DstMac":                 "0A:0B:0C:0D:0E:0F",
+				"SrcMac":                 "04:05:06:07:08:09",
+				"Duplicate":              false,
+				"Etype":                  2048,
+				"Packets":                123,
+				"Proto":                  6,
+				"SrcPort":                23000,
+				"DstPort":                443,
+				"Flags":                  0x100,
+				"TimeFlowStartMs":        someTime.UnixMilli(),
+				"TimeFlowEndMs":          someTime.UnixMilli(),
+				"Interface":              "eth0",
+				"AgentIP":                "10.11.12.13",
+				"PktDropBytes":           100,
+				"PktDropPackets":         10,
+				"PktDropLatestFlags":     0x200,
+				"PktDropLatestState":     "TCP_CLOSE",
+				"PktDropLatestDropCause": "SKB_DROP_REASON_TCP_CSUM",
+				"DnsLatencyMs":           someDuration.Milliseconds(),
+				"DnsId":                  1,
+				"DnsFlags":               0x8001,
+				"DnsFlagsResponseCode":   "FormErr",
+				"DnsErrno":               0,
+				"TimeFlowRttNs":          someDuration.Nanoseconds(),
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		// Test direct conversion
+		outDirect := ConvertToFLP(tt.flow)
+		assert.NotZero(t, outDirect["TimeReceived"])
+		delete(outDirect, "TimeReceived")
+		err := normalizeMap(outDirect)
+		require.NoError(t, err)
+		assert.Equalf(t, *tt.expected, outDirect, tt.name)
+
+		// Test pb conversion
```

Comment:
see here @msherif1234 : I am testing both FLP conversion AND protobuf conversion, with the bonus thatit makes sure the two are producing the same result (they're both compared against the same expected map)